File: cs25backend/src/main/java/edu/dongguk/cs25backend/dto/response/RestResponse.java
package edu.dongguk.cs25backend.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import edu.dongguk.cs25backend.exception.ErrorCode;
import jakarta.annotation.Nullable;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;


@Builder
@AllArgsConstructor
@Getter
public class RestResponse<T> {

    private final Boolean success;
    private final T data;
    @Builder.Default
    @JsonFormat(shape = JsonFormat.Shape.OBJECT)
    private ErrorCode error = null;

    public RestResponse(@Nullable T data) {
        this.success = true;
        this.data = data;
    }

    public static RestResponse<Object> errorResponse(ErrorCode error) {
        return new RestResponse<>(false, null, error);
    }


}

File: cs25server/src/main/kotlin/edu/dongguk/cs25server/dto/response/RestResponse.kt
package edu.dongguk.cs25server.dto.response

import edu.dongguk.cs25server.exception.ErrorCode

class RestResponse<T> (
    val success: Boolean,
    val data: T? = null,
    val error: ErrorCode? = null
) {

    constructor(data: T?) : this(true, data, null)

    companion object{
        fun errorResponse(error: ErrorCode?): RestResponse<Any> {
            return RestResponse(false, null, error)
        }
    }

}